force-app/main/default/classes/Collection Recipes/IterationRecipes.cls

```diff
@@ -14,11 +14,11 @@ public with sharing class IterationRecipes {
      **/
     public static Integer iterateOnAccountList(List<Account> accounts) {
         Integer employeeCount = 0;
-        Iterator<Account> it = accounts.iterator();
+        // Lists and Sets implement the Iterable interface in Apex
+        Iterable<Account> accountIterator = accounts;
```

Comment:
There is a design pattern I like using that is better, if accounts can be null.

```
for(Iterator<Account> acctIterator = accounts?.iterator();
      true == accounts?.hasNext();
) {
     Account acct = acctIterator.next();
     ...
}
```

The advantage of this syntax is if instead you did something like:

```
if(accounts != null) {
  for(Account acct : accounts) {
      ...
  }
}
``` 

You have a significantly higher cognitive complexity score.

And if you do not do the null check at all, you are risking a run-time exception.

Reply:
@docbill Thank you for the review. Please feel free to submit a PR or raise an issue. Happy to take the contributions.